### ISSUE TITLE
Can not render view when is primefaces on majora 2.3, when use PartialViewContext.setRenderAll(true) or use "@all" selector in primefaces component.

### ISSUE THREAD

**Issue by wklaczynski** [primefaces/primefaces#4666]
Can not render view when use PartialViewContext.setRenderAll(true) or use "@all"
 selector in update selector in primefaces components. With majora 2.2 all works fine. 

## 1) Environment
- PrimeFaces version: 6.2
- Application server + version: wildflay 15
- Affected browsers: firefox

**Comment by wklaczynski** [comment]
Also there is a problem with everything that requires render all. Functions that worked perfectly with primefaces 6.2 on the jsf 2.2 version did not work on jsf 2.3. The use of @form works perfectly but it is not always possible. 

            String ajaxscript = builder.init()
                    .source(sourceId)
                    .form(formId)
                    .event("scxmlhide")
                    .update(component, update != null ? update : "@all")
                    .process(component, process != null ? process : "@none")
                    .async(false)
                    .global(global != null ? Boolean.parseBoolean(global) : true)
                    .delay(null)
                    .timeout(0)
                    .partialSubmit(false, false, null)
                    .resetValues(false, false)
                    .ignoreAutoUpdate(true)
                    .onstart(null)
                    .onerror(null)
                    .onsuccess(null)
                    .oncomplete(null)
                    .buildBehavior(renderingMode);

https://github.com/wklaczynski/state-flow-faces/tree/master/state-flow-faces-demo/state-flow-prime-demo

https://github.com/wklaczynski/state-flow-faces/blob/master/state-flow-primefaces/src/main/java/org/ssoft/faces/prime/scxml/DialogInvoker.java

**Comment by tandraschko** [comment]
Please try 7.0 and provide a small as possible example without any state flow stuff.

**Comment by tandraschko** [comment]
Note: there also is an mojarra bug that doesnt render the resources in the head when using update @all. That might be the same bug

**Comment by wklaczynski** [comment]
Similar functions without the use of primafaces work on every JSF version. The problem always appears in the use of primefaces, PartialViewContext.setRenderAll (true), or any attempt to use @all, but only on the version 2.3.

            writer.startElement("script", component);
            writer.writeAttribute("type", "text/javascript", "type");
            writer.write("window.scxmltask = setTimeout(function(){");
            writer.write("clearTimeout(window.scxmltask);");
            writer.write("jsf.ajax.request(this,'scxmltask',{");
            writer.write("execute:'@none',render:'@all'");
            writer.write("})},");
            writer.write(String.valueOf(delay));
            writer.write(")");
            writer.endElement("script");

https://github.com/wklaczynski/state-flow-faces/blob/master/state-flow-fases-impl/src/main/java/org/ssoft/faces/impl/state/TimerEventProducerImpl.java

https://github.com/wklaczynski/state-flow-faces/blob/master/state-flow-fases-impl/src/main/java/org/ssoft/faces/impl/state/renderer/StateFlowScriptRenderer.java

The same function in primefaces implementation:

        AjaxRequestBuilder builder = requestContext.getAjaxRequestBuilder();
        String ajaxscript = builder.init()
                .source(sourceId)
                .form(formId)
                .event("scxmltask")
                .update(component, update)
                .process(component, "@all")
                .async(false)
                .global(false)
                .delay(null)
                .timeout(0)
                .partialSubmit(false, false, null)
                .resetValues(false, false)
                .ignoreAutoUpdate(true)
                .onstart(null)
                .onerror(null)
                .onsuccess(null)
                .oncomplete(null)
                .buildBehavior(renderingMode);

https://github.com/wklaczynski/state-flow-faces/blob/master/state-flow-primefaces/src/main/java/org/ssoft/faces/prime/TimerEventProducerImpl.java

without primefaces works fine on JSF 2.3. Use @all in alternative primefaces function works only JSF 2.2, Similarly with update @all in any primefaces component in the JSF 2.3 version.

**Comment by tandraschko** [comment]
I can check it tommorow but i need a simple cloned primefaces-test Project without stateflows etc.

Afair neither mojarra nor MyFaces implemented @all correctly. Only PrimeFaces does.

I think its the same error: https://github.com/eclipse-ee4j/mojarra/issues/4354

**Comment by wklaczynski** [comment]
Unfortunately, the same problem is in Primefaces 7.0.

I analyzed the problem of the render all function. Everything is fine when there is the original HeaderRenderer. When the PartialViewContextImpl setting for render all forces the rendering of the entire component tree. In the case when it is active, primefaces sets its own head renderer "org.primefaces.renderkit.HeadRenderer". The responce is missing heads of ComponentResource from viewRoot.

For page start with:


```
<html xmlns="http://www.w3.org/1999/xhtml"><head id="j_idt3"><link type="text/css" rel="stylesheet" href="/faces-demo/prime/faces/javax.faces.resource/theme.css?ln=primefaces-delta" /><link type="text/css" rel="stylesheet" href="/faces-demo/prime/faces/javax.faces.resource/demo-style.css?ln=css" /><link type="text/css" rel="stylesheet" href="/faces-demo/prime/faces/javax.faces.resource/components.css?ln=primefaces&amp;v=7.0" /><script type="text/javascript" src="/faces-demo/prime/faces/javax.faces.resource/jquery/jquery.js?ln=primefaces&amp;v=7.0"></script><script type="text/javascript" src="/faces-demo/prime/faces/javax.faces.resource/jquery/jquery-plugins.js?ln=primefaces&amp;v=7.0"></script><script type="text/javascript" src="/faces-demo/prime/faces/javax.faces.resource/core.js?ln=primefaces&amp;v=7.0"></script><script type="text/javascript" src="/faces-demo/prime/faces/javax.faces.resource/components.js?ln=primefaces&amp;v=7.0"></script><script type="text/javascript">if(window.PrimeFaces){PrimeFaces.settings.locale='pl_PL';PrimeFaces.settings.projectStage='Development';}</script>
        <title>Orders Demo</title></head><body>"


```


Partial all responce is start with:

```
<partial-response><changes><update id="javax.faces.ViewRoot"><![CDATA[<html xmlns="http://www.w3.org/1999/xhtml"><head id="j_idt3"><link type="text/css" rel="stylesheet" href="/faces-demo/prime/faces/javax.faces.resource/theme.css?ln=primefaces-delta" /><script type="text/javascript">if(window.PrimeFaces){PrimeFaces.settings.locale='pl_PL';PrimeFaces.settings.projectStage='Development';}</script>
        <title>Orders Demo</title></head><body>

```

**Comment by tandraschko** [comment]
Yup, thats the mojarra bug. Feel free to fix this on mojarra side.

**Comment by wklaczynski** [comment]
This is the correct operation. No matter if the render @all, head is not rendered during partial ajax submit for dynamic resources. The original jsf script dynamically consists of differences. Ajax script from primefaces does not do this. 
JSF selects dynamic resources as rendered in viewroot during restorephase, it always renders differences:

```
    @Override
    public void processEvent(ComponentSystemEvent event) throws AbortProcessingException {
        FacesContext context = event.getFacesContext();

        if (event instanceof PostRestoreStateEvent && context.getPartialViewContext().isPartialRequest()) {
            ResourceHandler resourceHandler = context.getApplication().getResourceHandler();

            for (UIComponent resource : getComponentResources(context)) {
                String name = (String) resource.getAttributes().get("name"); 
                String library = (String) resource.getAttributes().get("library"); 
                resourceHandler.markResourceRendered(context, name, library);
            }
        }


```

The problem is in "PrimeFaces.ajax":

 ```
         if(cfg.partialSubmit && processIds.indexOf('@all') === -1) {

        super.processEvent(event);
    }

```

**Comment by wklaczynski** [comment]
Maybe to solve your conflict between the majora, just do it when it's render all cleaning up the attribute in "RESOURCE_IDENTIFIER" FacesContext?

context.getAttributes().remove(RESOURCE_IDENTIFIER)

**Comment by tandraschko** [comment]
I think mojarra should just align their impl.
It doesnt make sense that PrimeFaces must always workaround such issues.

**Comment by wklaczynski** [comment]
Majora solves this problem in jsf.js. I solved it by doing it this way, so I have jsf 2.3 and primefaces. I can not imagine working without primefaces. I hope that you will solve this problem soon. This solution works great for me, so it's a closed topic for me.

https://github.com/wklaczynski/state-flow-faces/blob/master/state-flow-primefaces/src/main/java/org/ssoft/faces/prime/PrimeFlowPartialViewContext.java

**Comment by tandraschko** [comment]
As i said, it should be changed on mojarra side, not in PF.
